gbpservice/nfp/configurator/drivers/loadbalancer/v2/haproxy/haproxy_driver.py

```diff
@@ -0,0 +1,592 @@
+        member = n_data_models.Member.from_dict(
+            copy.deepcopy(member_dict)
+        )
+        ret = o_data_models.Member()
+        args = {
+            'id': member.id,
+            'project_id': member.tenant_id,
+            'pool_id': member.pool_id,
+            'ip_address': member.address,
+            'protocol_port': member.protocol_port,
+            'weight': member.weight,
+            'enabled': member.admin_state_up,
+            'subnet_id': member.subnet_id,
+            'operating_status': member.operating_status,
+        }
+        if member_dict.get('pool'):
+            pool = self.get_pool_octavia_model(member_dict.get('pool'))
+            args.update({
+                'pool': pool
+            })
+        ret = self._update(ret, args)
+        return ret
+
+    # Translate HealthMonitor neutron model dict to octavia model
+    def get_healthmonitor_octavia_model(self, hm_dict):
+        hm = n_data_models.HealthMonitor.from_dict(
+            copy.deepcopy(hm_dict)
+        )
+        ret = o_data_models.HealthMonitor()
+        args = {
+            'id': hm.id,
+            'project_id': hm.tenant_id,
+            'type': hm.type,
+            'delay': hm.delay,
+            'timeout': hm.timeout,
+            'rise_threshold': hm.max_retries,
+            'fall_threshold': hm.max_retries,
+            'http_method': hm.http_method,
+            'url_path': hm.url_path,
+            'expected_codes': hm.expected_codes,
+            'enabled': hm.admin_state_up
+        }
+        if hm_dict.get('pool'):
+            pool = self.get_pool_octavia_model(hm_dict.get('pool'))
+            args.update({
+                'pool': pool,
+                'pool_id': pool.id
+            })
+        ret = self._update(ret, args)
+        return ret
+
+
+class HaproxyLoadBalancerDriver(n_driver_base.LoadBalancerBaseDriver,
+                                LbGenericConfigDriver,
+                                base_driver.BaseDriver):
+    service_type = 'loadbalancerv2'
+    service_vendor = 'haproxy_lbaasv2'
+    # amphorae = {"loadbalancer_id": [o_data_models.Amphora(
+    #                                 lb_network_ip, id, status)]}
+    amphorae = {}
+
+    def __init__(self, plugin_rpc=None, conf=None):
+        super(HaproxyLoadBalancerDriver, self).__init__(plugin_rpc)
+
+        # Each of the major LBaaS objects in the neutron database
+        # need a corresponding manager/handler class.
+        #
+        # Put common things that are shared across the entire driver, like
+        # config or a rest client handle, here.
+        #
+        # This function is executed when neutron-server starts.
+        self.conf = conf
+        self.port = lbv2_constants.HAPROXY_AGENT_LISTEN_PORT
+
+        self.amphora_driver = HaproxyAmphoraLoadBalancerDriver()
+
+        self.load_balancer = HaproxyLoadBalancerManager(self)
+        self.listener = HaproxyListenerManager(self)
```

Comment:
Please use more self descriptive variable names. e.g self.listener to self.listener_manager